src/main/scala/com/databricks/labs/overwatch/pipeline/InitializerFunctionsUCE.scala

```diff
@@ -0,0 +1,210 @@
+package com.databricks.labs.overwatch.pipeline
+
+import com.databricks.dbutils_v1.DBUtilsHolder.dbutils
+import com.databricks.labs.overwatch.env.Database
+import com.databricks.labs.overwatch.utils.OverwatchScope._
+import com.databricks.labs.overwatch.utils._
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.SparkSession.setActiveSession
+import org.apache.spark.sql.catalyst.catalog.CatalogDatabase
+import org.apache.spark.sql.catalyst.dsl.expressions.DslSymbol
+import org.apache.spark.sql.functions.lit
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.util.SerializableConfiguration
+
+import scala.reflect.runtime.{universe => ru}
+
+class InitializerFunctionsUCE(config: Config, disableValidations: Boolean, isSnap: Boolean, initDB: Boolean)
+  extends InitializerFunctions (config: Config, disableValidations: Boolean, isSnap: Boolean, initDB: Boolean)
+    with SparkSessionWrapper {
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  override private[overwatch] def dataTargetIsValid(dataTarget: DataTarget): Boolean = {
+    val dbName = dataTarget.databaseName.getOrElse("overwatch").split("\\.").last
+    val rawDBLocation = dataTarget.databaseLocation.getOrElse(s"/user/hive/warehouse/${dbName}.db")
+    val dbLocation = PipelineFunctions.cleansePathURI(rawDBLocation)
+    val rawETLDataLocation = dataTarget.etlDataPathPrefix.getOrElse(dbLocation)
+    val etlDataLocation = PipelineFunctions.cleansePathURI(rawETLDataLocation)
+    val etlCatalogName = dataTarget.databaseName.get.split("\\.").head
+    var switch = true
+
+
+    spark.sessionState.catalogManager.setCurrentCatalog(etlCatalogName)
+    if (spark.catalog.databaseExists(dbName)) {
+      val dbMeta = spark.sessionState.catalog.getDatabaseMetadata(dbName)
```

Comment:
all this metadata stuff still works with UC? If so, great, just wanted to make sure you double checked.

Reply:
yes it does work with UC